src/nbytes.cpp

```diff
@@ -59,17 +59,16 @@ bool SwapBytes16(void *data, size_t nbytes) {
 #endif
 
   uint16_t temp;
-  uint8_t *ptr = reinterpret_cast<uint8_t *>(data);
-  for (size_t i = 0; i < nbytes; i += sizeof(uint16_t)) {
-    memcpy(&temp, &ptr[i], sizeof(uint16_t));
+  for (size_t i = 0; i < nbytes; i += sizeof(temp)) {
+    memcpy(&temp, &data[i], sizeof(temp));
     temp = BSWAP_2(temp);
-    memcpy(&ptr[i], &temp, sizeof(uint16_t));
+    memcpy(&data[i], &temp, sizeof(temp));
   }
 
   return true;
 }
 
-bool SwapBytes32(void *data, size_t nbytes) {
+bool SwapBytes32(char *data, size_t nbytes) {
   if (nbytes % sizeof(uint32_t) != 0) return false;
 
 #if defined(_MSC_VER)
```

Comment:
I am skeptical that this MS code is needed.

Reply:
Agreed. We can have SIMD support for this swap functions :-)

Reply:
> Agreed. We can have SIMD support for this swap functions :-)

Absolutely, although a single instruction can reverse 8 bytes so it is already very fast!!!